ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpBody.kt

```diff
@@ -121,3 +122,32 @@ public suspend fun parseHttpBody(
     ConnectionOptions.parse(headers["Connection"]),
     input, out
 )
+
+private fun isTransferEncodingChunked(transferEncoding: CharSequence): Boolean {
+    if (transferEncoding.equalsLowerCase(other = "chunked")) {
+        return true
+    }
+    if (transferEncoding.equalsLowerCase(other = "identity")) {
+        return false
+    }
+
+    var chunked = false
+    transferEncoding.split(",").forEach {
+        when (val name = it.trim().toLowerCase()) {
+            "chunked" -> {
+                if (chunked) {
+                    throw IllegalTransferEncodingException("Double-chunked TE is not supported: $transferEncoding")
+                }
+                chunked = true
+            }
+            "identity" -> {
+                // ignore this token
+            }
+            else -> throw IllegalTransferEncodingException("Unsupported transfer encoding $name")
+        }
+    }
+
+    return chunked
+}
+
+private class IllegalTransferEncodingException(override val message: String) : IllegalArgumentException(message)
```

Comment:
Could you tell me why do we need a new type of exception? Do we have the intention to catch it?